python/lsst/resources/gs.py

```diff
@@ -0,0 +1,291 @@
+# This file is part of lsst-resources.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# Use of this source code is governed by a 3-clause BSD-style
+# license that can be found in the LICENSE file.
+
+"""Accessing Google Cloud Storage resources."""
+
+from __future__ import annotations
+
+__all__ = ("GSResourcePath",)
+
+import contextlib
+import logging
+import re
+import tempfile
+from typing import IO, TYPE_CHECKING, Iterator, List, Optional, Set, Tuple, Union
+
+try:
+    import google.cloud.storage as storage
+    from google.cloud.exceptions import NotFound
+except ImportError:
+    storage = None
```

Comment:
I guess due to auth considerations it isn't possible to fall back to `s3://` in this case?

Reply:
For the IDF the notebook setup would handle that because we have both credentials set up, although notebook environments already have google-cloud-storage installed. It would require some changes to the `__new__` method to understand that a failure to import should change the scheme to s3 but it seems like in the general case that would be more confusing than telling people to install google-cloud-storage.